### PR TITLE
Fix validation of resources (cpu, memory, storage) for limit range types

### PR DESCRIPTION
**Pull request by aveshagarwal**
Fix validation of standard container resources only for pod and container when validating limit range Max.

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
So we would need to touch every call of `validateContainerResourceName` in this function...

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2061
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2066
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2080
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2085
https://github.com/kubernetes/kubernetes/blob/master/pkg/api/validation/validation.go#L2092

Ideally, we would have a validation function that takes the LimitType as an argument and call that instead and have that delegate to either call `validateResourceName` or `validateContainerResourceName` appropriately.

Once you have that create a unit test in validation_test for TestValidateLimitRange that passes a different LimitType in for example, "mythirdparty.com/something" and be sure it validates fine.

**Comment by aveshagarwal** [comment]
@derekwaynecarr updated this PR. PTAL.

**Comment by aveshagarwal** [comment]
@derekwaynecarr if you are ok, I can send it to kube upstream and we can continue discussion there, anything is fine.

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -2058,12 +2058,20 @@ func ValidateLimitRange(limitRange *api.LimitRange) field.ErrorList {
 		maxLimitRequestRatios := map[string]resource.Quantity{}
 
 		for k, q := range limit.Max {
-			allErrs = append(allErrs, validateContainerResourceName(string(k), idxPath.Child("max").Key(string(k)))...)
+			if limit.Type == api.LimitTypePod || limit.Type == api.LimitTypeContainer {
```
Comment (derekwaynecarr) [inline pkg/api/validation/validation.go]: can we just have a function like this so its not needing to be repeated everywhere in the if blocks?

```
func validateLimitRangeResourceName(limitType api.LimitType, resourceName string) {
 switch limitType {
   case api.LimitTypePod, api.LimitTypeContainer:
     return validateContainerResourceName()
   default:
     return validateResourceName()
 }
}
```

File: pkg/api/validation/validation_test.go [hunk 2]
```diff
@@ -3864,6 +3864,14 @@ func getResourceList(cpu, memory string) api.ResourceList {
 	return res
 }
 
+func getNonStandardContainerResourceList(storage string) api.ResourceList {
```
Comment (derekwaynecarr) [inline pkg/api/validation/validation_test.go]: call this `getStorageResourceList(storage string) api.ResourceList` for clarity.

**Comment by derekwaynecarr** [comment]
just make those updates then we can open upstream.  I think we also want something that validates limit type is a valid qualified name if not standard to upstream.  similar to `validateResourceName`

**Comment by derekwaynecarr** [comment]
cc me on the pr upstream, so I can review.

**Comment by aveshagarwal** [comment]
@derekwaynecarr sure. will open PR soon.

**Pull request closed without merging**